server/hostname.cpp

```diff
@@ -1,35 +1,13 @@
 #include "hostname.h"
 
-#include <systemd/sd-bus.h>
+#include <string>
+#include <unistd.h>
 
 std::string hostname()
 {
-	sd_bus * bus;
-	if (sd_bus_default_system(&bus) < 0)
-		return "";
-
-	for (auto property: {"PrettyHostname", "StaticHostname", "Hostname"})
-	{
-		char * hostname = nullptr;
-		sd_bus_error error = SD_BUS_ERROR_NULL;
-
-		if (sd_bus_get_property_string(bus, "org.freedesktop.hostname1", "/org/freedesktop/hostname1", "org.freedesktop.hostname1", property, &error, &hostname) < 0)
-		{
-			sd_bus_error_free(&error);
-			continue;
-		}
-
-		sd_bus_error_free(&error);
-		if (hostname && strcmp(hostname, ""))
-		{
-			std::string s = hostname;
-			free(hostname);
-			sd_bus_unref(bus);
-			return s;
-		}
-	}
-
-	sd_bus_unref(bus);
-
-	return "Unknown";
+  char buf[256];
```

Comment:
Can we make the `gethostname`/dbus switch an ifdef?

PrettyHostname will allow non-latin characters and uppercase.

Reply:
there's now a separate `hostname.cpp` and `hostname-sd.cpp`, the latter will be included by cmake if systemd is available.